src/testing/tree_construction/generator.rs

```diff
@@ -0,0 +1,110 @@
+use crate::html5::node::{Node, NodeData, NodeTrait, NodeType};
+use crate::html5::node::{MATHML_NAMESPACE, SVG_NAMESPACE, XLINK_NAMESPACE, XMLNS_NAMESPACE};
+use crate::html5::parser::document::DocumentHandle;
+
+/// Generates a tree output that can be used for matching with the expected output
+pub struct TreeOutputGenerator {
+    document: DocumentHandle,
+}
+
+impl TreeOutputGenerator {
```

Comment:
This outputs a given parsed tree (document) into the same format as found in the fixture tests.